src/utils_wmi.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * collectd - src/utils_wmi.h
+ * Copyright (c) 2018  Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#ifndef UTILS_WMI_H
+#define UTILS_WMI_H
+
+#include <oaidl.h>
+
+#define LIST_NODE_DECL(node_type) node_type _node
+#define LIST_NEXT_DECL(node_type) node_type *_next
+#define LIST_NODE(nodeptr) (nodeptr->_node)
+#define LIST_NEXT(nodeptr) (nodeptr->_next)
+#define LIST_HEAD(list) (list)
+
+#define LIST_TYPE(type) list_##type##_t
+
+#define LIST_DECL_TYPE(node_type)                                              \

```

Comment:
Hrm, I wonder if we actually need all of these "magic" defines. Anyway, please wrap each define with multiple statements into do {} while (0) -- that way it's less likely to cause problems, for example when using like:

    if (something)
      LIST_FOO()

Reply:
Done. I was able to remove a lot of the defines, including the list types. I did keep the LIST_INSERT_FRONT and LIST_FREE but moved them to wmi.c